pkgs/top-level/python-packages.nix

```diff
@@ -1007,6 +1007,8 @@ in {
 
   backports_functools_lru_cache = callPackage ../development/python-modules/backports_functools_lru_cache { };
 
+  backports_functools_lru_cache_121 = callPackage ../development/python-modules/backports_functools_lru_cache_121 { };
```

Comment:
Even if `arrow` really does require a very specific version of `lru_cache` this version should not be exposed in `python-packages.nix`.

Ideally `arrow` should be patched to not require a very explicit version but if it really does this override should be moved to the `arrow` directory.

Reply:
Additional versions of packages are not permitted in `python-packages.nix` *at all* as they cause collisions, so also not inside expressions. Applications that are defined *outside* `python-packages.nix` can override the package set to insert whatever version they need.